server/handlers/repos/unlink.go

```diff
@@ -0,0 +1,103 @@
+package repos
+
+import (
+	"encoding/json"
+	"log"
+	"net/http"
+
+	"github.com/hn275/envhub/server/api"
+	"github.com/hn275/envhub/server/database"
+	"github.com/hn275/envhub/server/gh"
+	"github.com/jackc/pgerrcode"
+	"github.com/jackc/pgx/v5/pgconn"
+)
+
+// returns 201 on success, no body
+// { "message": "err" } otherwise
+func Unlink(w http.ResponseWriter, r *http.Request) {
+	if r.Method != http.MethodDelete {
+		api.NewResponse(w).Status(http.StatusMethodNotAllowed).Done()
+		return
+	}
+
+	user, err := api.NewContext(r).User()
+	if err != nil {
+		api.NewResponse(w).Status(http.StatusForbidden).Error(err.Error())
+		return
+	}
+
+	// only `repo.ID` and `repo.FullName` are sent
+	var repo database.Repository
+	var partialData map[string]interface{}
+
+	// Decode the JSON into a map
+	if err := json.NewDecoder(r.Body).Decode(&partialData); err != nil {
+		api.NewResponse(w).Status(http.StatusBadRequest).Error(err.Error())
+		return
+	}
+
+	// Populate the specific fields in the 'repo' struct
+	if id, ok := partialData["repoID"].(uint64); ok {
+		repo.ID = id
+	}
+	if fullName, ok := partialData["repoName"].(string); ok {
+		repo.FullName = fullName
+	}
+
+	// Now, 'repo' should contain the decoded values
+	// log.Println("Received repo:", repo)
```

Comment:
here is some Go superpower for ya
```go
var repo struct {
    RepoID uint64 `json:"repoID"`
    FullName string `json:"fullName"`
}

if err := json.NewDecoder(r.Body).Decode(&repo); err != nil {
    // ...
}
```

The field names have to be exported/public for the json package to marshal data properly.